R/reportMif.R

```diff
@@ -0,0 +1,55 @@
+#' Create mif file for model run
+#'
+#' The mif file contains reporting variables for the given model run.
+#'
+#' @param path character, path to the run
+#' @param file character, path of mif file. If \code{NULL}, default file name
+#'   in run folder will be used. If \code{FALSE}, no file is written and the mif
+#'   data is returned instead.
+#'
+#' @author Robin Hasse
+#'
+#' @importFrom reportBrick convGDX2MIF
+#' @export
+
+reportMif <- function(path, file = NULL) {
+
+  if (!dir.exists(path)) {
+    stop("The given directory does not exist: ", path)
+  }
+
+  # recognised file names for model output (in descending order of priority)
+  gdxNames <- c("output.gdx")
+
+  # find gdx file
+  gdx <- head(file.path(path, gdxNames), 1)
+  if (is.null(gdx)) {
+    stop("No gdx file found for reporting in given path: ", path)
+  }
+
+  # read config
+  configFile <- file.path(path, "config", "config.yaml")
+  config <- readConfig(configFile)
```

Comment:
Wouldn't it make sense to use the readDirect option of readConfig in this case? As "config.yaml" is the fully assembled configuration file? I know we discussed whether or not this option makes sense at all, but we decided to keep it and I would think this is a case where it applies.

Reply:
You are right. I changed the function call accordingly. This made me realise that I unconsciously relied on the non-direct reading of the config for old runs to report mifs as they miss the `reportingTemplate` parameter. So I added a new argument that enables users to pass it instead of using the config setting.